### DISCUSSION TITLE
SQL issue - FOR each returns b'undefined'

### DISCUSSION THREAD

**Discussion by vibhubithar** [emqx/emqx#9855] [General]
**payload**
{
   "timestamp":"2023-01-28T07:02:10.662Z",
   "id":1,
   "name":"Demo",
   "reid":"Demo/SC",
   "rename":"SC",
   "counts":{
      "vehicle":1
   },
   "objects":[
      {
         "ca":"vehicle",
         "c":{
            "a":315,
            "b":387,
            "c":65.0,
            "d":41.5
         },
         "v":[
            0.2745151953475066,
            3.6774511006086135,
            0.0
         ]
      },
      {
         "ca":"vehicle",
         "c":{
            "a":318,
            "b":399,
            "c":65.0,
            "d":41.5
         },
         "v":[
            0.2745151953475066,
            3.6774511006086135,
            0.0
         ]
      }
   
]
}

SQL 1: this one works fine on the payload

select
  payload.objects[1].ca,
  payload.objects[1].v,
  payload.objects[1].c
FROM "$bridges/mqtt:S2-SC"

output:
got post request:  b'{"objects":[{"v":[0.5212939901600879,5.903456416075176,0.0],"c":{"a":344,"b":361,"c":60.333333333333336,"d":41.25},"ca":"vehicle"}]}'

**SQL2** : this one is returning undefined

foreach
   payload.objects as po
do
   po.ca,
   po.v,
   po.c
from "$bridges/mqtt:S2-SC"

output: 
got post request:  b'undefined'
10.0.0.172 - - [28/Jan/2023 11:10:43] "POST / HTTP/1.1" 200 -

**Q1: why foreach is not working, when i can get individual array elements?**

**Q2 : how do i test the SQL to see output on a payload before putting it in a rule?**

**Comment by lafirest** [comment]
@vibhubithar 
In the dashboard, you can test the SQL with payload by enabling the `Test SQL` toggle button
![image](https://user-images.githubusercontent.com/22971499/215814161-ad1c9828-5e38-43f6-9336-b96773bbb6f0.png)
And I have tested your SQL with your payload, I think everything is right, I guess maybe there were some problems in your `Action Handler`.
